func/checkbinary.py

```diff
@@ -19,38 +21,43 @@ def getbinary(gametype):
 
         #Checking
         binary = ""
+        
+        if gametype != "epic":
+            executable = "gogdl "
+        else:
+            executable = "legendary "
 
-        if os.path.exists("/opt/Heroic/resources/app.asar.unpacked/build/bin/linux") == True:
+        heroic_base_path = None
 
-            if gametype != "epic":
-                binary = "/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl "
-            else:
-                binary = "/opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary "
+        # Attempt to find heroic on path
+        detected_heroic_path = shutil.which('heroic')
+        if detected_heroic_path:
+            # Follow any symlinks to the real path of heroic
+            detected_heroic_path = os.path.realpath(detected_heroic_path)
+            if os.path.exists(detected_heroic_path):
+                heroic_base_path = os.path.dirname(detected_heroic_path)
+        elif os.path.exists("/opt/Heroic"): # Default to /opt/Heroic
+            heroic_base_path = "/opt/Heroic"
+        elif configpath.is_flatpak or os.path.exists("/app/bin/heroic"): #System or Flatpak-env path
+            heroic_base_path = "/app/bin/heroic"
+        
+        if heroic_base_path:
+
+            heroic_resources_path = os.path.join(heroic_base_path, resources_bin_path)
+
+            if os.path.exists(heroic_resources_path):
```

Comment:
There's an issue here in case of Heroic Flatpak. 

The path check on line 48 will fail as `/app/bin/heroic/resources/app.asar.unpacked/build/bin/linux/<executable>` is only present in the Flatpak environment.

So `binary` will remain as an empty string and the executable path won't be displayed in the bash launch script.

Reply:
I just tested this on my VM where I've got Heroic Flatpak installed and the script fails to launch any game as the binary path is empty. As a workaround, I commented this check and it worked perfectly fine.

Maybe you can keep this check for the non-Flatpak versions of Heroic?

Reply:
Thanks for the clarification! I wasn't as familiar with this path but that makes sense. I am going to remove the extra check since the other branches actually already check `os.path.exists()` before continuing. I also updated the unit test to reflect this. 

Talking about this made me think of a question: should `configpath.is_flatpak` be checked first in this method? While unlikely, someone could have Heroic installed natively and as a Flatpak, and the method here would return the system installation of Heroic, but the paths being returned for the paths in configpath will always be for the Flatpak. Depending on the expected behavior, I could add a unit test to check this edge case too.

Also, would it be okay to update build.yml as part of this PR to run unit tests automatically?

Reply:
Good point. I guess you reorder the checking accordingly.

Reply:
@redromnon I pushed a change yesterday to prefer the Flatpak, but realized handling that edge case by making the Flatpak check first could also cause other users to be impacted by #102.

I have pushed another change to attempt to fix #102 by detecting when the Heroic Flatpak is installed.
Let me know if you'd prefer I save that for another PR and I can undo it and 072f576 for now to keep the behavior closer to what's in main today.

Reply:
@afwolfe I guess it'll be better to fix that issue in another PR. I had completely forgotten about that issue.